server/src/main/java/org/opensearch/index/remote/RemoteMigrationClusterStateUtils.java

```diff
@@ -0,0 +1,191 @@
+            logger.info(
+                "Index {} does not have remote store based index settings but all primary shards and STARTED replica shards have moved to remote enabled nodes. Applying remote store settings to the index",
+                index
+            );
+            Settings.Builder indexSettingsBuilder = Settings.builder().put(currentIndexSettings);
+            MetadataCreateIndexService.updateRemoteStoreSettings(indexSettingsBuilder, segmentRepoName, tlogRepoName);
+            indexMetadataBuilder.settings(indexSettingsBuilder);
+            indexMetadataBuilder.settingsVersion(1 + indexMetadata.getVersion());
+        } else {
+            logger.debug("Index {} does not satisfy criteria for applying remote store settings", index);
+        }
+    }
+
+    /**
+     * Returns <code>true</code> iff all the below conditions are true:
+     * - All primary shards are in {@link ShardRoutingState#STARTED} state and are in remote store enabled nodes
+     * - No replica shard in {@link ShardRoutingState#RELOCATING} state
+     * - All {@link ShardRoutingState#STARTED} replica shards are in remote store enabled nodes
+     *
+     * @param indexRoutingTable    current {@link IndexRoutingTable} from cluster state
+     * @param discoveryNodes       set of discovery nodes from cluster state
+     * @param currentIndexSettings current {@link IndexMetadata} from cluster state
+     * @return <code>true</code> or <code>false</code> depending on the met conditions
+     */
+    public static boolean needsRemoteIndexSettingsUpdate(
+        IndexRoutingTable indexRoutingTable,
+        DiscoveryNodes discoveryNodes,
+        Settings currentIndexSettings
+    ) {
+        assert currentIndexSettings != null : "IndexMetadata for a shard cannot be null";
+        if (indexHasRemoteStoreSettings(currentIndexSettings) == false) {
+            boolean allPrimariesStartedAndOnRemote = indexRoutingTable.shardsMatchingPredicate(ShardRouting::primary)
+                .stream()
+                .allMatch(shardRouting -> shardRouting.started() && discoveryNodes.get(shardRouting.currentNodeId()).isRemoteStoreNode());
+            List<ShardRouting> replicaShards = indexRoutingTable.shardsMatchingPredicate(shardRouting -> shardRouting.primary() == false);
+            boolean noRelocatingReplicas = replicaShards.stream().noneMatch(ShardRouting::relocating);
+            boolean allStartedReplicasOnRemote = replicaShards.stream()
+                .filter(ShardRouting::started)
+                .allMatch(shardRouting -> discoveryNodes.get(shardRouting.currentNodeId()).isRemoteStoreNode());
+            return allPrimariesStartedAndOnRemote && noRelocatingReplicas && allStartedReplicasOnRemote;
+        }
+        return false;
+    }
+
+    /**
+     * Updates the remote store path strategy metadata for the index when it is migrating to remote.
+     * This should be run only when the first primary copy moves over from docrep to remote.
```

Comment:
We are not checking for this and marking this settings on first relocation/shard started/failed post making the cluster `mixed mode` . This behavior is quite proactive and harmless . Please call this out .

Reply:
Updated the javadoc with this info